cmd/miner/cpuminer.go

```diff
@@ -194,18 +198,28 @@ func (m *CPUMiner) solve(ctx context.Context) {
 		jobID := m.miner.work.jobID
 		m.miner.workMtx.RUnlock()
 
-		if m.solveBlock(ctx, headerB, target, ticker) {
-			// Send the request.
+		switch m.solveBlock(ctx, headerB, target) {
+		case true:
+			// Send a submit work request.
 			worker := fmt.Sprintf("%s.%s", m.miner.config.Address,
 				m.miner.config.User)
 			id := m.miner.nextID()
 			req := pool.SubmitWorkRequest(&id, worker, jobID,
 				m.workData.extraNonce2, m.workData.nTime, m.workData.nonce)
 			m.workCh <- req
 
-			// Stall to prevent mining too quickly.
-			time.Sleep(time.Second)
+		case false:
+			select {
+			case <-ctx.Done():
+				m.miner.wg.Done()
+				return
+			default:
+				// Non-blocking receive fallthrough.
+			}
 		}
+
+		// Stall to prevent mining too quickly.
```

Comment:
This seems to have moved. Previously it was only executed when `solveBlock` returned true, now it is executed on every loop iteration. Is this correct?

Reply:
good catch, didn't need to move it.

Reply:
Good catch, didn't need to move it.